cs/test/core/GuidConversionTests.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace UnitTest
+{
+    using System;
+    using Bond;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// Summary description for GuidConversionTests
+    /// </summary>
+    [TestFixture]
+    public class GuidConversionTests
+    {
+        [Test]
+        public void ImplicitConversion()
+        {
+            var systemGuid = Guid.NewGuid();
+            var bondGuid = (GUID)systemGuid;
+
+            // The Datas should be populated if it converted correctly
+            Assert.AreNotEqual(bondGuid.Data1, 0);

```

Comment:
I'd be more confident in the test if we used a static GUID and asserted the values of the Data1..Data4 members explicitly to make sure that each representation of the GUID is correct, instead of just that we can round trip a GUID.

(When you do this, you'll likely need to make an assumption about the endianess of the memory layout of the test machine. That's fine. Bonus points for encoding such an assumption in a test case.)